src/Swift.Runtime/src/Swift/SwiftSet.cs

```diff
@@ -0,0 +1,174 @@
+    {
+        var witnessTable = ProtocolWitnessTable.GetOrThrow<Element, ISwiftHashable>();
+        return TypeMetadata.Cache.GetOrAdd(typeof(SwiftSet<Element>), _ => SwiftSetPInvokes.PInvoke_getMetadata(TypeMetadataRequest.Complete, ElementTypeMetadata, witnessTable));
+    }
+
+    static TypeMetadata ElementTypeMetadata
+    {
+        get => TypeMetadata.GetTypeMetadataOrThrow<Element>();
+    }
+
+    static ISwiftObject ISwiftObject.NewFromPayload(SwiftHandle handle)
+    {
+        return new SwiftSet<Element>(handle);
+    }
+
+    IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
+    {
+        var metadata = SwiftObjectHelper<SwiftSet<Element>>.GetTypeMetadata();
+        unsafe
+        {
+            fixed (void* _payloadPtr = &_variant)
+            {
+                metadata.ValueWitnessTable->InitializeWithCopy((void*)swiftDest, (void*)_payloadPtr, metadata);
+            }
+        }
+        return swiftDest;
+    }
+
+    /// <summary>
+    /// Gets the protocol conformance descriptor for the given type.
+    /// </summary>
+    /// <typeparam name="TProtocol"></typeparam>
+    /// <returns></returns>
+    static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescriptor<TProtocol>()
+        where TProtocol : class
+    {
+        if (!_protocolConformanceSymbols.TryGetValue(typeof(TProtocol), out var symbolName))
+        {
+            throw new SwiftRuntimeException($"Attempted to retrieve protocol conformance descriptor for type SwiftSet and protocol {typeof(TProtocol).Name}, but no conformance was found.");
+        }
+        return ProtocolConformanceDescriptor.LoadFromSymbol("/usr/lib/swift/libswiftCore.dylib", symbolName);
+    }
+
+    /// <summary>
+    /// Constructs a new SwiftSet from the given handle.
+    /// </summary>
+    unsafe SwiftSet(SwiftHandle handle)
+    {
+        this._variant = *(Variant*)(handle);
+    }
+
+    /// <summary>
+    /// Constructs a new empty SwiftSet.
+    /// </summary>
+    public SwiftSet()
+    {
+        var witnessTable = ProtocolWitnessTable.GetOrThrow<Element, ISwiftHashable>();
+        _variant = SwiftSetPInvokes.Init(ElementTypeMetadata, witnessTable);
+    }
+
+    /// <summary>
+    /// Gets the number of elements in the set.
+    /// </summary>
+    public unsafe int Count
+    {
+        get
+        {
+            var witnessTable = ProtocolWitnessTable.GetOrThrow<Element, ISwiftHashable>();
+            return (int)SwiftSetPInvokes.Count(_variant, ElementTypeMetadata, witnessTable);
+        }
+    }
+}
+
+internal static class SwiftSetPInvokes
+{
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sShMa")]
+    public static extern TypeMetadata PInvoke_getMetadata(TypeMetadataRequest request, TypeMetadata typeMetadata, ProtocolWitnessTable witnessTable);
+
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sS2hyxGycfC")]
+    public static extern Variant Init(TypeMetadata typeMetadata, ProtocolWitnessTable witnessTable);
```

Comment:
We should probably name the first parameter `elementTypeMetadata` - to make it more descriptive what it needs to be.